apps/ex_nvr_web/lib/ex_nvr_web/controllers/api/remote_storage_controlller.ex

```diff
@@ -0,0 +1,83 @@
+defmodule ExNVRWeb.API.RemoteStorageController do
+  @moduledoc false
+
+  use ExNVRWeb, :controller
+
+  action_fallback ExNVRWeb.API.FallbackController
+
+  plug :authorization_plug
+  plug :remote_storage_plug, [field_name: "id"] when action in [:update, :show, :delete]
+
+  import ExNVRWeb.Controller.Helpers
+  import ExNVR.Authorization
+
+  alias ExNVR.RemoteStorages
+  alias ExNVR.Model.RemoteStorage
+  alias Plug.Conn
+
+  def authorization_plug(%Conn{} = conn, _opts) do
+    user = conn.assigns.current_user
+
+    case authorize(user, :remote_storage, :any) do
+      :ok ->
+        conn
+
+      {:error, :unauthorized} ->
+        conn
+        |> forbidden()
+        |> Conn.halt()
+    end
+  end
+
+  def remote_storage_plug(%Conn{} = conn, opts) do
+    field_name = Keyword.get(opts, :field_name, "remote_storage_id")
+    remote_storage_id = conn.path_params[field_name]
+
+    case RemoteStorages.get(remote_storage_id) do
+      %RemoteStorage{} = remote_storage ->
+        Logger.metadata(remote_storage_id: remote_storage.id)
```

Comment:
No need for this, all the endpoints have this `id` in their path, so it's easy to know from the logs which remote storage is requested.